src/binary-writer.cc

```diff
@@ -880,17 +878,29 @@ static Result write_module(Context* ctx, const Module* module) {
 
     char desc[100];
     begin_custom_section(ctx, WABT_BINARY_SECTION_NAME, LEB_SECTION_SIZE_GUESS);
-    write_u32_leb128(&ctx->stream, 1, "function name type");
-    begin_subsection(ctx, "function name subsection", LEB_SECTION_SIZE_GUESS);
-    write_u32_leb128(&ctx->stream, module->funcs.size(), "num functions");
-    for (size_t i = 0; i < module->funcs.size(); ++i) {
-      const Func* func = module->funcs[i];
-      write_u32_leb128(&ctx->stream, i, "function index");
-      wabt_snprintf(desc, sizeof(desc), "func name %" PRIzd, i);
-      write_str(&ctx->stream, func->name.start, func->name.length,
-                PrintChars::Yes, desc);
+
+    size_t named_functions = 0;
+    for (const auto func: module->funcs) {
```

Comment:
I've been avoiding using `auto` except for iterators and such. I think it makes it a bit clearer. Also, I believe the Chromium style guide suggests not using auto for pointers, though I can't quite remember why.